reindent.d

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env rdmd
+
+/* Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ */
+
+module reindent;
+
+import std.range.primitives;
+
+/**
+ * Reindents an input range of lines.
+ *
+ * Each line is assumed to consist of an initial run of spaces that are n
+ * multiples of inputIndent. The output lines are constructed by substituting
+ * this initial run of spaces with n multiples of outputIndent instead.
+ *
+ * The primary usage is intended to be for reindenting space-indented code.
+ *
+ * Note that the input is assumed to be already properly-indented. This
+ * function does not attempt to parse or pretty-print the input.
+ *
+ * Params:
+ *  range = Input range of lines to be reindented.
+ *  inputIndent = The number of spaces to be counted as a single level of
+ *      indentation in the input.
+ *  outputIndent = The number of spaces per indentation level to be used in the
+ *      output.
+ *
+ * Returns: An input range of lines.
+ */
+auto reIndent(R)(R range, size_t inputIndent, size_t outputIndent)
+    if (isInputRange!R && is(ElementType!R : const(char)[]))
+{
+    import std.algorithm.iteration : map;
+    import std.array;
+    import std.regex;
+
+    auto reSplitInitialSpace = regex(`^( +)(.*)`, "s");
+    auto reReindent = regex(" ".replicate(inputIndent));
```

Comment:
This is probably a little overkill, just looking at `m.captures[1].length` would suffice.